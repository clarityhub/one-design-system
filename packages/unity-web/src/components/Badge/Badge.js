import React from 'react';
import isPropValid from '@emotion/is-prop-valid';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { node, oneOf } from 'prop-types';
import colors from '@clarityhub/unity-core/lib/colors';
import { opacify } from '@clarityhub/unity-core/lib/utilities/color';
import { variants } from '@clarityhub/unity-core/lib/typography';

const Badge = styled('div', {
	shouldForwardProp: prop => isPropValid(prop) && !['type'].includes(prop),
})`
    overflow: visible;
    text-transform: none;
    display: inline-block;

    ${variants.button.string}

    font-weight: 600;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    border: 0 solid transparent;
    padding: 0.1rem 0.4rem;
    font-size: 0.8rem;
    line-height: 1.2;
    border-radius: 2rem;
    position: relative;
    text-transform: uppercase;

    ${({ type }) => {
		switch (type) {
		case 'primary':
			return css`
                    color: ${colors.primary.default};
                    background-color: ${opacify(colors.primary.default, 0.25)};
                `;
		case 'success':
			return css`
                    color: ${colors.success.default};
                    background-color: ${opacify(colors.success.default, 0.25)};
                `;
		case 'danger':
			return css`
                    color: ${colors.danger.default};
                    background-color: ${opacify(colors.danger.default, 0.25)};
                `;
		case 'notification':
			return css`
                    color: ${colors.white.default};
                    background-color: ${colors.notification.default};
                    border-color: ${colors.notification.default};
                `;
		case 'default':
		default:
			return css`
                    color: ${colors.black.default};
                    background-color: ${colors.muted.default};
                    border-color: ${colors.muted.default};
                `;
		}
	}}
`;

/**
 * Hack for docz and react-docgen
 */
const BadgePropTypes = () => <div />;
BadgePropTypes.propTypes = {
	children: node,
	type: oneOf(['primary', 'success', 'danger', 'notification', 'default']),
};
Badge.propTypes = BadgePropTypes.propTypes;
export { BadgePropTypes };

export default Badge;
