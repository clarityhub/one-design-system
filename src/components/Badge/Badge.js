import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { oneOf } from 'prop-types';

import colors from '../../theme/colors';
import { opacify } from '../../utilities/color';
import { variants } from '../../theme/fonts';

const Badge = styled.div`
    overflow: visible;
    text-transform: none;
    display: inline-block;

    ${variants.button}

    font-weight: 600;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    border: 0 solid transparent;
    padding: 0.1rem 0.4rem;
    font-size: 0.8rem;
    line-height: 1.2;
    border-radius: 2rem;
    position: relative;
    text-transform: uppercase;

    ${({ type }) => {
        switch (type) {
            case 'primary':
                return css`
                    color: ${colors.primary};
                    background-color: ${opacify(colors.primary, 0.3)};
                `;
            case 'success':
                return css`
                    color: ${colors.success};
                    background-color: ${opacify(colors.success, 0.3)};
                `;
            case 'danger':
                return css`
                    color: ${colors.danger};
                    background-color: ${opacify(colors.danger, 0.3)};
                `;
            case 'notification':
                return css`
                    color: ${colors.white};
                    background-color: ${colors.notification};
                    border-color: ${colors.notification};
                `;
            case 'default':
            default:
                return css`
                    color: ${colors.black};
                    background-color: ${colors.muted};
                    border-color: ${colors.muted};
                `;
        }
    }}
`;

/**
 * Hack for docz and react-docgen
 */
const BadgePropTypes = () => <div />;
BadgePropTypes.propTypes = {
    type: oneOf(['primary', 'success', 'danger', 'notification', 'default']),
};
Badge.propTypes = BadgePropTypes;
export { BadgePropTypes };

export default Badge;
