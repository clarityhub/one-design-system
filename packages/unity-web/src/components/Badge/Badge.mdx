---
menu: Components
name: Badges
---

import { Playground, PropsTable } from 'docz';
import Badge, { BadgePropTypes } from './Badge';
import Link from '../Link';

# Badges

## \<Badge /\> component

```js
import Badge from '@clarityhub/unity-web/lib/components/Badge';
```

## Prop Types

<PropsTable of={BadgePropTypes} />

## Examples

### Sizes

<Playground>
    <Badge>1</Badge>
    <br />
    <br />
    <Badge>new</Badge>
    <br />
    <br />
    <Badge>1 new</Badge>
</Playground>

### Colors
<Playground>
    <Badge type="default">10</Badge>
    <br />
    <br />
    <Badge type="primary">10</Badge>
    <br />
    <br />
    <Badge type="success">10</Badge>
    <br />
    <br />
    <Badge type="danger">10</Badge>
    <br />
    <br />
    <Badge type="notification">10</Badge>
    <br />
    <br />
</Playground>

### As Links
<Playground>
    <Link type="text" href="#"><Badge type="default">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="primary">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="success">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="danger">10</Badge></Link>
    <br />
    <br />
    <Link type="text" href="#"><Badge type="notification">10</Badge></Link>
    <br />
    <br />
</Playground>
