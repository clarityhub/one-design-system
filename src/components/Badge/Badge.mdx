---
menu: Components
name: Badges
---

import { Playground, PropsTable } from 'docz';
import Badge, { BadgePropTypes } from './Badge';

# Badges

## \<Badge /\> component

```js
import Badge from '@clarityhub/one/components/Badge';
```

## Prop Types

<PropsTable of={BadgePropTypes} />

## Examples

### Sizes

<Playground>
    <Badge>1</Badge>
    <br />
    <br />
    <Badge>new</Badge>
    <br />
    <br />
    <Badge>1 new</Badge>
</Playground>

### Colors
<Playground>
    <Badge type="default">10</Badge>
    <br />
    <br />
    <Badge type="primary">10</Badge>
    <br />
    <br />
    <Badge type="success">10</Badge>
    <br />
    <br />
    <Badge type="danger">10</Badge>
    <br />
    <br />
    <Badge type="notification">10</Badge>
    <br />
    <br />
</Playground>
