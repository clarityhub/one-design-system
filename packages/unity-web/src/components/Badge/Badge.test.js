import React from 'react';
import renderer from 'react-test-renderer';

import Badge from './Badge';

describe('<Badge />', () => {
	it('renders different badge types', () => {
		const node = (
			<div>
				<Badge type="default">Default</Badge>
				{' '}
				<Badge type="primary">Primary</Badge>
				{' '}
				<Badge type="success">Success</Badge>
				{' '}
				<Badge type="danger">Danger</Badge>
				{' '}
				<Badge type="notification">Notification</Badge>
			</div>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('renders different badge sizes and outlines', () => {
		const node = (
			<div>
				<Badge type="default" square>Default</Badge>
				{' '}
				<Badge type="primary" size="large">Primary</Badge>
				{' '}
				<Badge type="success" outline>Success</Badge>
				{' '}
				<Badge type="danger" square outline>Danger</Badge>
				{' '}
				<Badge type="notification">Notification</Badge>
			</div>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});