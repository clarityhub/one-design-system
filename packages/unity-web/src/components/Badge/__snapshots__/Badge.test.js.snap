// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders different badge sizes and outlines 1`] = `
.emotion-8 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #FFF;
  background-color: #FF6A5C;
  border-color: #FF6A5C;
}

.emotion-0 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #333;
  background-color: #F1F1F1;
  border-color: #C2C2C2;
  cursor: pointer;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  border-radius: 5px;
}

.emotion-2 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #566ad2;
  background-color: rgba(86,106,210,0.25);
  border-color: rgba(86,106,210,0.25);
  font-size: 1rem;
}

.emotion-4 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #46C161;
  background-color: rgba(70,193,97,0.25);
  border-color: rgba(70,193,97,0.25);
  background: transparent;
  border-width: 1px;
  border-style: solid;
}

.emotion-6 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #f5365c;
  background-color: rgba(245,54,92,0.25);
  border-color: rgba(245,54,92,0.25);
  cursor: pointer;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  border-radius: 5px;
  background: transparent;
  border-width: 1px;
  border-style: solid;
}

<div>
  <div
    className="emotion-0 emotion-1"
  >
    Default
  </div>
   
  <div
    className="emotion-2 emotion-1"
    size="large"
  >
    Primary
  </div>
   
  <div
    className="emotion-4 emotion-1"
  >
    Success
  </div>
   
  <div
    className="emotion-6 emotion-1"
  >
    Danger
  </div>
   
  <div
    className="emotion-8 emotion-1"
  >
    Notification
  </div>
</div>
`;

exports[`<Badge /> renders different badge types 1`] = `
.emotion-0 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #333;
  background-color: #F1F1F1;
  border-color: #C2C2C2;
}

.emotion-2 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #566ad2;
  background-color: rgba(86,106,210,0.25);
  border-color: rgba(86,106,210,0.25);
}

.emotion-4 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #46C161;
  background-color: rgba(70,193,97,0.25);
  border-color: rgba(70,193,97,0.25);
}

.emotion-6 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #f5365c;
  background-color: rgba(245,54,92,0.25);
  border-color: rgba(245,54,92,0.25);
}

.emotion-8 {
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  font-weight: 600;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  border: 0 solid transparent;
  padding: 0.1rem 0.4rem;
  font-size: 0.8rem;
  line-height: 1.2;
  border-radius: 2rem;
  position: relative;
  text-transform: uppercase;
  color: #FFF;
  background-color: #FF6A5C;
  border-color: #FF6A5C;
}

<div>
  <div
    className="emotion-0 emotion-1"
  >
    Default
  </div>
   
  <div
    className="emotion-2 emotion-1"
  >
    Primary
  </div>
   
  <div
    className="emotion-4 emotion-1"
  >
    Success
  </div>
   
  <div
    className="emotion-6 emotion-1"
  >
    Danger
  </div>
   
  <div
    className="emotion-8 emotion-1"
  >
    Notification
  </div>
</div>
`;
